addons/misra.py

```diff
@@ -2726,11 +2728,17 @@ def reportError(self, location, num1, num2):
             if self.severity:
                 cppcheck_severity = self.severity
 
-            cppcheckdata.reportError(location, cppcheck_severity, errmsg, 'misra', errorId, misra_severity)
+            this_violation = '{}-{}-{}'.format(location.file, location.linenr, ruleNum)
```

Comment:
This will (I think) silence multiple warnings on the same line? It is not uncommon (at least in my experience with misra) to have multiple warnings of the same kind of the same line.

For example, this should give two violations of rule 13.5 but with the proposed changes will only give one (untested, I'm not on the computer now):

```
if (a && f++ < 10 && g++ < 10) {}
```

This would still give two warnings
```
if (a &&
    f++ < 10 &&
    g++ < 10) {}
```
Which is a bit inconsistent.

Reply:
Good catch.  I've added the column value to the key.